File: 10 - Python/Exercícios/04 - Vetores/Exercício 03.py
#Exercício - Alturas

soma = 0
menor = 0

N = int(input("Quantas pessoas serão digitadas? "))

nome = [0] * N
idade = [0] * N
altura = [0] * N

for i in range(N):

    print(f"Dados da {i+1}º pessoa:")
    nome[i] = input("Nome: ")
    idade[i] = int(input("Idade: "))
    altura[i] = float(input("Altura: "))
    print("")

    soma = soma + altura[i]

    if idade[i] < 16:
        menor = menor + 1

media = soma / N
porc = menor * 100 / N

print("Altura média:", round(media,2))
print("Pessoas com menos de 16 anos:", porc, "%")

for i in range(N):
    
    if idade[i] < 16:
        print(nome[i])

File: 06 - C/Exercícios/04 - Vetores/Exercício 03.c
//Exercício - Alturas

#include <stdio.h>
#include <string.h>

void limpar_entrada() {
    char c;
    while ((c = getchar()) != '\n' && c != EOF) {}
}

void ler_texto(char *buffer, int length) {
    fgets(buffer, length, stdin);
    strtok(buffer, "\n");    
}

int main(){

    int i, N, menor;
    double soma, media, porc;

    soma = 0;
    menor = 0;

    printf("Quantas pessoas serao digitadas? ");
    scanf("%d", &N);

    char nome[N][50];
    int idade[N];
    double altura[N];


    for(i = 0; i < N; i++){

        printf("\nDados da %da pessoa:\n", i+1);
        printf("Nome: ");
        limpar_entrada();
        ler_texto(nome[i], 50);
        printf("Idade: ");
        scanf("%d", &idade[i]);
        printf("Altura: ");
        scanf("%lf", &altura[i]);

        soma = soma + altura[i];

        if(idade[i] < 16){
            menor = menor + 1;
        }
    }

    media = (double) soma / N;
    porc = (double) menor / N * 100;
    
    printf("\nAltura media: %.2lf", media);
    printf("\nPessoas com menos de 16 anos: %.1lf%%\n", porc);
    
    for(i = 0; i < N; i++){
        if(idade[i] < 16){
            printf("%s\n", nome[i]);
        }
    }

    return 0;

}